### PR TITLE
Fixed #32421 -- Made admindocs ModelDetailView show model cached properties.

### PR DESCRIPTION
**Pull request by ramonsaraiva**
https://code.djangoproject.com/ticket/32421

### CODE REVIEW

File: django/contrib/admindocs/views.py [hunk 1]
```diff
@@ -261,11 +262,17 @@ def get_context_data(self, **kwargs):
...
                 if isinstance(func, property):
                     fields.append({
-                        'name': func_name,
+                        'name': '%s (property)' % func_name,
+                        'data_type': get_return_data_type(func_name),
+                        'verbose': verbose or ''
+                    })
+                elif isinstance(func, cached_property):
+                    fields.append({
+                        'name': '%s (cached property)' % func_name,
```
Comment (felixxm) [inline django/contrib/admindocs/views.py:275]: I don't see much value in added prefixes: `(cached property)`, `(property)`, and putting them in the `name` column can be misleading. We don't include prefixes for `classmethods`, `staticmethods` etc. 

If it's important for someone they can always add such information to the docstring.

File: django/contrib/admindocs/views.py [hunk 2]
```diff
@@ -261,11 +262,17 @@ def get_context_data(self, **kwargs):
                 verbose = verbose and (
                     utils.parse_rst(cleandoc(verbose), 'model', _('model:') + opts.model_name)
                 )
-                # Show properties and methods without arguments as fields.
+                # Show properties and cached_properties and methods without arguments as fields.
```
Comment (felixxm) [inline django/contrib/admindocs/views.py:265]: Wrap comments at 79 chars.

File: django/contrib/admindocs/views.py [hunk 3]
```diff
@@ -250,7 +251,7 @@ def get_context_data(self, **kwargs):
         methods = []
         # Gather model methods.
         for func_name, func in model.__dict__.items():
-            if inspect.isfunction(func) or isinstance(func, property):
+            if inspect.isfunction(func) or isinstance(func, (property, cached_property)):
```
Comment (felixxm) [inline django/contrib/admindocs/views.py:254]: Please alphabetize.
```suggestion
            if inspect.isfunction(func) or isinstance(func, (cached_property, property)):
```

**Review by felixxm — COMMENTED** [review]
@ramonsaraiva Thanks for this patch :+1: Please add also a `versionchanged` admonition to the [docs](https://docs.djangoproject.com/en/3.1/ref/contrib/admin/admindocs/#model-reference), e.g.
```
.. versionchanged:: 4.0

    Older versions don't display model cached properties.
```

File: django/contrib/admindocs/views.py (hunk 1 above)
Comment (ramonsaraiva) [reply django/contrib/admindocs/views.py:275]: Agree, removing those.

**Comment by felixxm** [comment]
@ramonsaraiva Thanks :+1: I added release notes.

**Comment by ramonsaraiva** [comment]
Thank you!

**Pull request merged**